lib/pure/strutils.nim

```diff
@@ -2699,6 +2697,8 @@ proc addf*(s: var string, formatstr: string, a: varargs[string, `$`]) {.
   var i = 0
   var num = 0
   while i < len(formatstr):
+    template invalidFormatString() =
+      raise newException(ValueError, "invalid format string: i: " & $i & "\nprefix:\n" & formatstr[0..<i] & "\nsuffix:\n" & formatstr[i..^1])
```

Comment:
The ``.noinline`` proc produces better code.

Reply:
good point; I'm moving this change to another PR and will fix that